src/index.ts

```diff
@@ -254,23 +245,8 @@ class TaskInfo extends AsyncResource implements Task {
   ) {
     super('Tinypool.Task', { requireManualDestroy: true, triggerAsyncId })
     this.callback = callback
-    this.task = task
+    this.task = fillTransferList(task, transferList)
```

Comment:
This PR also adds nested movable support to the `pool.run` method. ☝️ 

Note that only 1 level of nesting is checked. In other words, doing this 👇 is not supported:
```ts
pool.run({ a: { b: Tinypool.move(b) } })
```